.github/workflows/test.yml

```diff
@@ -3,14 +3,15 @@ name: Run DTS tests
 on:
   pull_request:
     branches:
-      - 'main'
+      - "main"
 jobs:
   build-dts:
     uses: ./.github/workflows/build.yml
     with:
       cacheless: false
-  run-tests:
-    name: Run DTS tests
+
+  prepare-for-tests:
```

Comment:
Each job is run in different session so most likely even detached processes would be killed between jobs.
You need to start ipxe server and qemu in the same job you are using it. As matrix creates multiple jobs you probably need to start qemu/ipxe before each `robot` step

Reply:
Moved here: 051caa1d67b115128723802a638aeb2cc97d31a9